### ISSUE TITLE
Azure Stack flavor names are case sensitive

### ISSUE THREAD

**Issue by bsquizz** [ManageIQ/manageiq-providers-azure#252]
During Azure Stack testing, I tried to deploy a a private image and hit this error:

```
[----] E, [2018-05-07T12:27:03.581936 #5157:121f114] ERROR -- : Q-task_id([miq_provision_1]) MIQ(ManageIQ::Providers::Azure::CloudManager::Provision#provision_error) [[Azure::Armrest::BadRequestException]: The requested VM size standard_a1 is not available in the current region. The sizes available in the current region are: Basic_A0,Basic_A1,Basic_A2,Basic_A3,Basic_A4,Standard_A0,Standard_A1,Standard_A2,Standard_A3,Standard_A4,Standard_A5,Standard_A6,Standard_A7,Standard_D1,Standard_D2,Standard_D3,Standard_D4,Standard_D11,Standard_D12,Standard_D13,Standard_D14,Standard_D1_v2,Standard_D2_v2,Standard_D3_v2,Standard_D4_v2,Standard_D5_v2,Standard_D11_v2,Standard_D12_v2,Standard_D13_v2,Standard_D14_v2,Standard_DS1,Standard_DS2,Standard_DS3,Standard_DS4,Standard_DS11,Standard_DS12,Standard_DS13,Standard_DS14,Standard_DS1_v2,Standard_DS2_v2,Standard_DS3_v2,Standard_DS4_v2,Standard_DS5_v2,Standard_DS11_v2,Standard_DS12_v2,Standard_DS13_v2,Standard_DS14_v2. Find out more on the available VM sizes in each region at https://aka.ms/azure-regions.] encountered during phase [start_clone_task]
```

Requested vm size was 'standard_a1', and Standard_A1 is in the list. It appears that the flavor name is case sensitive for the Azure Stack APIs.

We could get around this by only storing the uid as `flavor.name.downcase` and keeping the name as-is (see https://github.com/ManageIQ/manageiq-providers-azure/compare/master...bsquizz:flavor_names), however I'm not sure if altering the display name will affect anything existing customers are doing with Azure providers. If that's the case, we may need to store another attribute (i.e. `actual_name`) alongside the downcase'd `name` and have `prepare_for_clone_task` in provision/cloning.rb use the `actual_name` for `vmSize`

**Comment by djberg96** [comment]
Quite silly, I would consider that a bug in the Azure API. It shouldn't care. They're already somewhat inconsistent on the issue, e.g. it ignores case for resource groups.

Let me think about how we should handle it.

**Comment by djberg96** [comment]
@bsquizz Any chance you could create a BZ for this so I can reference it?

**Comment by bsquizz** [comment]
https://bugzilla.redhat.com/show_bug.cgi?id=1594013

**Comment by djberg96** [comment]
Should be fixed with #272
